packages/a11y-base/src/list-mixin.js

```diff
@@ -117,8 +118,13 @@ export const ListMixin = (superClass) =>
       if (this._observer) {
         this._observer.flush();
       }
-      const firstItem = this.querySelector('[tabindex="0"]') || (this.items ? this.items[0] : null);
-      this._focusItem(firstItem);
+      const firstItem = this.querySelector('[tabindex="0"]');
+      if (firstItem && !isElementHidden(firstItem)) {
+        this._focusItem(firstItem);
```

Comment:
Ideally this should also be done through the `this._focus` method for a consistent result.

Reply:
Thanks, done. Also added a test to check that items other than focused one get `tabindex` set to `-1`.